src/FloatingButton/FloatingButton.tsx

```diff
@@ -0,0 +1,198 @@
+        this.setState({ active: false, display: true });
+        setTimeout(() => this.setState({ display: false }), animate ? 200 : 0);
+      } else {
+        this.setState({ active: true, display: true });
+      }
+    }
+    if (onClick && !active) {
+      onClick();
+    }
+  };
+
+  render() {
+    const {
+      additionalButtons,
+      animate,
+      bgColor,
+      className,
+      closeLabel,
+      label,
+      positionX,
+      positionY,
+      size,
+    } = this.props;
+    const { active, display } = this.state;
+
+    return (
+      <FlexBox
+        className={classnames(
+          cssClass.CONTAINER,
+          cssClass.propStyle(positionX),
+          cssClass.propStyle(positionY),
+          className,
+        )}
+        style={{
+          ...this.horizontalStyle(),
+          ...this.verticalStyle(),
+        }}
+      >
+        <div className={classnames(cssClass.SUB_ELEMENT, className)}>
+          <Button
+            className={cssClass.BUTTON}
+            onClick={this.mainButtonHandler}
+            value={
+              active
+                ? closeLabel || (
+                    <div>
+                      <span className="fa fa-times" /> Close
+                    </div>
+                  )
+                : label
+            }
+            size={size || "regular"}
+            style={{
+              borderRadius: "500px",
+              ...(bgColor
+                ? {
+                    backgroundColor: bgColor,
+                    borderColor: bgColor,
+                  }
+                : {}),
+              ...(active
+                ? {
+                    backgroundColor: "#474c5e", // @neutral_dark_gray
+                    borderColor: "#474c5e",
+                    color: "white",
+                  }
+                : {}),
+            }}
+          />
+        </div>
+        {additionalButtons &&
+          additionalButtons.map((button, i) => (
+            <div
+              className={classnames(
+                cssClass.SUB_ELEMENT,
+                !active && animate ? cssClass.inactive(positionY) : "",
+                button.className,
+              )}
+              key={i}
+            >
+              <Button
+                className={cssClass.BUTTON}
+                onClick={button.onClick}
+                value={button.label}
+                size={size || "regular"}
+                style={{
+                  borderRadius: "500px",
+                  ...(bgColor
+                    ? {
+                        backgroundColor: bgColor,
+                        borderColor: bgColor,
+                      }
+                    : {}),
+
+                  // With mutliple clicks within transition window, we may reach active=true, display=false edge case.
+                  // We should only remove from DOM when inactive and not persisting display for transition out.
+                  display: active || display ? "inline" : "none",
```

Comment:
We can likely define this style in the stylesheet as well. We just need to apply a CSS class when the active prop is specified and then style accordingly. Happy to chat through what this might look like in person!

In short, I don't think you need the `style` prop for anything in this PR except for the `positionX` and `positionY`

Reply:
I played around with it and couldn't find a way to work without using the style prop. Putting it in the css class caused it to not animate. My guess is this has something to do with the order of operations of style application (seems the style prop takes precedence). In general, what I found is that transition does not mix well with display. Most resources I found online suggested using visibility: none instead of display, but I wasn't satisfied with that because I don't people clicking on buttons accidentally. Open to suggestions for better ways to do this but this is the best I've got.

Reply:
Ah yes, the `style` prop takes highest precedence. But there are still ways to get your CSS to take precedence over the `<Button>`'s internal styles. I've found that, to get styles to work on the `<Button>`, you often need to use a compound selector:

```
button.FloatingButton--button
```

Reply:
That did the trick!